packages/network-controller/src/NetworkController.ts

```diff
@@ -909,22 +917,26 @@ export class NetworkController extends BaseControllerV2<
    * latest block has a `baseFeePerGas` property, then updates state
    * appropriately.
    *
+   * @param networkClientId - The networkClientId to fetch the correct provider against which to check 1559 compatibility.
    * @returns A promise that resolves to true if the network supports EIP-1559
    * , false otherwise, or `undefined` if unable to determine the compatibility.
    */
-  async getEIP1559Compatibility() {
-    if (!this.#ethQuery) {
+  async getEIP1559Compatibility(networkClientId?: NetworkClientId) {
```

Comment:
Do we want this method to continue to exist? Or should consumers simply read it off state?

Reply:
I guess currently this is the time when we actually update this state... But I think we're going to have to rethink when this state gets updated now that it's a collection.

Reply:
I think we need to refactor `lookupNetwork` to take in a `networkClientId` and refresh the status and EIP1559 compatibility info if it hasn't been updated in x amount of time?

Reply:
Ticket for this ☝️ here: https://app.zenhub.com/workspaces/devex-63bee08a4e3b9d001108416e/issues/gh/metamask/metamask-planning/1090

Reply:
this is a public method unless i'm missing something?

Reply:
Oh this one is a public method, my bad. Okay this change makes sense, nothing much we can do about that for now at least.